### DISCUSSION TITLE
GSOC IDEA : Proposal for Unified CLI Tool and Library for IHR

### DISCUSSION THREAD

**Discussion by MAVRICK-1** [InternetHealthReport/gsoc#38] [Ideas]
# Proposal for Unified CLI Tool and Library for IHR Data Analysis & Exploration

The current frontend interface lacks the capability to comprehensively display specific data and is unsuitable for conducting data analytics and exploration . It is essential to develop a dedicated library tailored for data analysis purposes to overcome these limitations. Relying solely on the frontend for analysis tasks is inadequate, especially for individuals with limited knowledge beyond Neo4j. Hence , there is a need for an IYP library that provides Data of each node.

A specialized library for IYP (Internet Yellow Pages) data analysis and data explorationwould be advantageous, providing users with the tools necessary to conduct comprehensive data analytics tasks efficiently. Such a library would enable:

- Data selection and indexing
- Data manipulation
- Data cleaning
- Data visualization
- Data transformation
- Advanced operations
- Integration with other libraries

You can explore the prototype at https://github.com/MAVRICK-1/IYP_Lib. It functions as a dedicated library where CLI features can be integrated seamlessly.

Combining existing IHR libraries—abondance, route-origin-validator, and IYP—into a unified CLI tool and library is proposed. This consolidation would introduce new Python modules and streamline data retrieval processes, making them faster and more organized. For users, having a single, comprehensive CLI tool would greatly enhance usability and manageability.

For example, the CLI tool could be used as follows:

```bash
$ ihr rov 8.8.8.0/24 15169 --rpki_archive 2023/10/01
$ ihr iyp 2497
$ ihr hegemony 2500 --start "2018-09-15 00:00" --end "2018-09-15 23:59"
```

This also provides output in the form of JSON, CSV, or DataFrame as required by the user and will also print the data in the form of tables so that the user can see it. These commands illustrate the versatility of the tool, allowing users to retrieve specific data related to ROV (Route Origin Validation), IYP (Internet Yellow Pages), and Hegemony for analysis and exploration.

In conclusion, the development of a unified CLI tool and library for IHR data analysis would significantly enhance the capabilities of users in handling diverse datasets and analytical tasks effectively.

@romain-fontugne @m-appel what your views on this ?

**Comment by dpgiakatos** [comment]
This is a duplicate of #16. Please continue the discussion there.

**Comment by MAVRICK-1** [comment]
> This is a duplicate of #16. Please continue the discussion there.

@dpgiakatos sir this discussion is slightly different. Here I am talking about unified library along with a IYP library and some add on for the IYP library

**Comment by AdamLucife** [comment]
@MAVRICK-1 Hi, I got one question? you get the data for IYP through API?

**Reply by MAVRICK-1** [reply]
For my proposal, I query from knowledge graph.

**Comment by SAHELISAHAA** [comment]
@MAVRICK-1 Can you please provide the resources , To know for this ?

**Comment by dpgiakatos** [comment]
Hello, as previously mentioned, this proposal duplicates the one already put forth in comment https://github.com/orgs/InternetHealthReport/discussions/16#discussioncomment-8498478, within discussion #16. Let's make an effort to avoid redundant discussions to maintain clarity and streamline communication. Thank you for your cooperation!
